### PR TITLE
Cleanup tmp files when exiting

### PR DESCRIPTION
**Pull request by apiraino**
hey @jtheoof 

This PR ensures that swappy deletes temp files when exiting or when piping `grim` and pressing ESC and thus having stdin empty.

What do you think?

### CODE REVIEW

**Comment by apiraino** [comment]
@jtheoof pinging for a review on this one (no rush, when you have time)

**Comment by jtheoof** [comment]
I'm not sure to understand the need for this @apiraino. The variable serves to store the file name location and is freed using `g_free` when needed. Did you see a specific issue?

**Comment by apiraino** [comment]
hi @jtheoof thanks for getting back.

yes, I do experience an issue: tmp files are not deleted in all cases (example: when the screenshot is not saved). If you can't reproduce I can recompile your upstream version and retry (but I'm quite sure I did so)

**Comment by apiraino** [comment]
@jtheoof could you please check again this PR?

It fixes a minor (but annoying) papercut, temp files in `/tmp` are never deleted. I've tried solving this by manually calling a function to delete the temporary file the application is about to exit (with or without saving the file).

How to reproduce: run `grim -g "$(slurp)" - | swappy -f -`, select an area, then press <esc> key, now check `/tmp` for the temporary file for a leftover.

thanks

**Comment by jtheoof** [comment]
Sorry about the late reply @apiraino I'm gonna try to reproduce this locally first. Will give you more feedback soon.

File: src/application.c [hunk 1]
```diff
@@ -256,6 +257,21 @@ void copy_clicked_handler(GtkWidget *widget, struct swappy_state *state) {
   clipboard_copy_drawing_area_to_selection(state);
 }
 
+void delete_tmpfile(struct swappy_state *state) {
+  if (g_unlink(state->file_str) != 0)
```
Comment (jtheoof) [inline src/application.c:261]: This works when using `-f -` command but will also delete a non temporary file when using:

```
./build/swappy -f docs/images/screenshot-1.0.0.png
```

I'm working on a better fix. Thanks for bringing this up @apiraino

**Review by jtheoof — CHANGES_REQUESTED** [review]
This will delete existing files from the filesystem when not using the `stdin` shortcut.

I'll work a better fix.

**Comment by jtheoof** [comment]
Thanks for bringing this up @apiraino. I merged #81 with a slightly better fix for the `-f` option. 

Can you try it out and let me know if you find issues?

**Comment by apiraino** [comment]
Sorry about my clumsy code fiddling :-) and thanks for providing the right fix

**Pull request closed without merging**